ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -194,12 +209,42 @@ export class WorkflowsList extends BasePage<RouteComponentProps<any>, State> {
         );
     }
 
-    private fetchWorkflows(namespace: string, selectedPhases: WorkflowPhase[], selectedLabels: string[], pagination: Pagination): void {
+    private lastMonth() {
+        const dt = new Date();
+        dt.setMonth(dt.getMonth() - 1);
+        dt.setHours(0, 0, 0, 0);
+        return dt;
+    }
+
+    private nextDay() {
+        const dt = new Date();
+        dt.setDate(dt.getDate() + 1);
+        dt.setHours(0, 0, 0, 0);
+        return dt;
+    }
+
+    private parseTime(dateStr: string) {
+        if (dateStr != null) {
+            return new Date(dateStr);
+        }
+    }
+
+    private fetchWorkflows(namespace: string, selectedPhases: WorkflowPhase[], selectedLabels: string[], minStartedAt: Date, maxStartedAt: Date, pagination: Pagination): void {
         if (this.listWatch) {
             this.listWatch.stop();
         }
         this.listWatch = new ListWatch(
-            () => services.workflows.list(namespace, selectedPhases, selectedLabels, pagination),
+            () =>
+                services.workflows.list(namespace, selectedPhases, selectedLabels, pagination).then(x => {
+                    if (x.items && x.items.length > 20) {
+                        this.appContext.apis.notifications.show({
```

Comment:
I don’t think you need to do this. The filter looks pretty fast to me.

Reply:
I understand.